packages/react-form/src/utilities.ts

```diff
@@ -174,15 +174,32 @@ export function getValues<T extends FieldBag>(fieldBag: T) {
 }
 
 export function getDirtyValues<T extends FieldBag>(fieldBag: T) {
-  return Object.entries(fieldBag).reduce(
-    (acc, [fieldName, field]: [string, Field<any>]) => {
+  const reduceFn = (acc, [fieldName, field]: [string, Field<any>]) => {
+    if (Array.isArray(field)) {
+      const dirtyArray = field
+        .map((dirtyField) => Object.entries(dirtyField).reduce(reduceFn, {}))
+        .filter((dirtyField) => Object.keys(dirtyField).length);
       return {
         ...acc,
-        ...(field.dirty ? {[fieldName]: field.value} : {}),
+        ...(dirtyArray.length ? {[fieldName]: dirtyArray} : {}),
       };
-    },
-    {},
-  );
+    }
+
+    if (!isField(field)) {
```

Comment:
Great questions and thanks for taking a look at this! To answer your review comment, I'm not sure if this would break anyones existing flow. This should still function exactly as the previous `getDirtyValues` util did, the previous tests pass and it seems to be backwards compatible. I'm open to adding more tests and I just saw the `/snapit` functionality, so open to taking a snapshot and testing in some projects where this util is used.

For this comment, you are correct that this test is functionally checking if the field is an object. I chose `!isField` as a field under the hood really is just an object haha, so any other means of checking if it is an object will yield true even when it is a field object and not just a plain object. Looking at the type definitions and testing with fields that are not just strings, anything passed into this helper should be a `FieldBag` (object collection) and if its not TS will be unhappy and it will break. Not sure how defensive we need to be here so welcome to ideas there too!

<img width="346" alt="image" src="https://user-images.githubusercontent.com/29992628/202562087-fa5fadfa-e7b5-46db-9200-d65adededc44.png">

Reply:
Interesting, I had added a comment here and it didn't go along with the review 🤔 

I meant to say I was worried something that wasn't a `Field` but was something we were not expecting, like `null` (_because Javascript is weird_) could allow us to get into this `if` block. Upon further inspection, I realized my fear was unfounded, though.